File: SharedPreference/app/src/main/java/com/example/sharedpreference/MyApplication.java
package com.example.sharedpreference;

import android.app.Application;

import com.example.sharedpreference.sharedpreference.DataLocalManager;

public class MyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        DataLocalManager.init(getApplicationContext());
    }
}

File: MyHome/app/src/main/java/com/example/myhome/MyApplication.kt
package com.example.myhome

import android.app.Application
import com.example.myhome.data.repository.MyDataLocal
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        MyDataLocal.init(applicationContext)
    }
}